File: Doubly_linked_list/DoublyLinkedListInsertBeginning.java
// File Name: DoublyLinkedListInsertBeginning.java
// Program to insert a node at the beginning of a doubly linked list

class Node {
    int data;
    Node prev;
    Node next;

    Node(int data) {
        this.data = data;
        this.prev = null;
        this.next = null;
    }
}

public class DoublyLinkedListInsertBeginning {
    static Node head = null;

    // Function to insert at beginning
    static void insert_beg(int value) {
        Node newNode = new Node(value);
        newNode.next = head;

        if (head != null)
            head.prev = newNode;

        head = newNode;
    }

    public static void main(String[] args) {
        insert_beg(10);
    }
}

File: Doubly_linked_list/doubly_linked_list_insert_beginning.c
// File Name: doubly_linked_list_insert_beginning.c
// Program to insert a node at the beginning of a doubly linked list

#include <stdio.h>
#include <stdlib.h>

// Definition of node
struct node {
    int data;
    struct node *prev;
    struct node *next;
};

struct node *head = NULL;

// Function to insert at beginning
void insert_beg(int value) {
    struct node *newnode = (struct node *)malloc(sizeof(struct node));

    newnode->data = value;
    newnode->prev = NULL;
    newnode->next = head;

    if (head != NULL)
        head->prev = newnode;

    head = newnode;
}

int main() {
    insert_beg(10);
    return 0;
}
